plugins/destination/s3/client/write.go

```diff
@@ -66,10 +72,37 @@ func (c *Client) WriteTableBatch(ctx context.Context, table *schema.Table, data
 	return nil
 }
 
-// sanitizeJSONKeys replaces all invalid characters in JSON keys with underscores.
-// It does the replacement in-place, modifying the original object. This is required
-// for compatibility with Athena.
-func sanitizeJSONKeys(obj any) {
+// sanitizeRecordJSONKeys replaces all invalid characters in JSON keys with underscores. This is required
+// for compatibility with Athena. It returns a new record, and the old one is released.
+func sanitizeRecordJSONKeys(mem memory.Allocator, record arrow.Record) (arrow.Record, error) {
+	newRecordBuilder := array.NewRecordBuilder(mem, record.Schema())
+	defer record.Release()
```

Comment:
We can leave this out I think to make things consistent.

Reply:
Updated